File: 0092-reverse-linked-list-ii/0092-reverse-linked-list-ii.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseBetween(ListNode* head, int left, int right) {
        
        ListNode* temp = head;
        ListNode* prev1 = head;
        int z = left;
        while(z>1){
            prev1=temp;
            temp=temp->next;
            z--;
        }
        // cout<<temp->val<<endl;
        ListNode* prev = temp;
        ListNode* num = temp;
        ListNode* num1 = num->next;
        int k = right-left;
        while(k>0){
            ListNode*num2 = num1->next;
            num1->next=num;
            num=num1;
            num1=num2;
            k--;
        }
        // cout<<num->val<<endl;
        prev1->next=num;
        prev->next=num1;
        
        if(left==1) return num;
        
        return head;
        
    }
};

File: 0092-reverse-linked-list-ii/0092-reverse-linked-list-ii.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseBetween(self, head, m, n):
        """
        :type head: ListNode
        :type m: int
        :type n: int
        :rtype: ListNode
        """
        dummy = ListNode(0)
        dummy.next = head
        pre = dummy

        for _ in range(m - 1):
            pre = pre.next

        cur = pre.next

        for _ in range(n - m):
            temp = pre.next
            pre.next = cur.next
            cur.next = cur.next.next
            pre.next.next = temp

        return dummy.next
